include/red_black_tree.h

```diff
@@ -128,6 +141,231 @@ class RedBlackTree : public BinarySearchTree<K, V> {
         }
         _root->color = ColorType::BLACK;
     }
+
+    // rb-tree 的 remove 也是一个 2-3-4 树的 remove 过程
+    // 首先，按照正常 bst 的删除方法，删除 u，但是先得记住 u 的颜色
+    void remove(Node* u) {
+        assert(u != nullptr);
+        auto star_color = u->color;
+        Node* x = nullptr; // star 节点（打星号标记的节点）
+        Node* xp = nullptr; // x 的爸爸
```

Comment:
1. Shall we name the variables as more meaningful?

Reply:
xp means x's parent.

Reply:
...